cargo/.devcontainer/Dockerfile

```diff
@@ -10,47 +9,42 @@ ARG CONTAINER_GROUP=esp
 ARG ESP_BOARD=all
 ARG GITHUB_TOKEN
 
-# Install dependencies
+ARG RUST_INSTALL_DIR=/usr/local/cargo
+
+# Install dependencies needed for esp-idf + clang for riscv rust
 RUN apt-get update \
-    && apt-get install -y git curl gcc clang ninja-build libudev-dev unzip xz-utils\
-    python3 python3-pip python3-venv libusb-1.0-0 libssl-dev pkg-config libpython2.7 \
-    && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts
+    && apt-get install -y flex bison gperf python3-pip python3-venv cmake ninja-build ccache dfu-util clang \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/* /tmp/library-scripts
 
 # Set users
 RUN adduser --disabled-password --gecos "" ${CONTAINER_USER}
 USER ${CONTAINER_USER}
 WORKDIR /home/${CONTAINER_USER}
 
-# Install rustup
-RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- \
-    --default-toolchain none -y --profile minimal
-
-# Update envs
-ENV PATH=${PATH}:/home/${CONTAINER_USER}/.cargo/bin
-
 # Install extra crates
-RUN ARCH=$($HOME/.cargo/bin/rustup show | grep "Default host" | sed -e 's/.* //') && \
-    curl -L "https://github.com/esp-rs/espup/releases/latest/download/espup-${ARCH}" -o "${HOME}/.cargo/bin/espup" && \
-    chmod u+x "${HOME}/.cargo/bin/espup" && \
-    curl -L "https://github.com/esp-rs/espflash/releases/latest/download/cargo-espflash-${ARCH}.zip" -o "${HOME}/.cargo/bin/cargo-espflash.zip" && \
-    unzip "${HOME}/.cargo/bin/cargo-espflash.zip" -d "${HOME}/.cargo/bin/" && \
-    rm "${HOME}/.cargo/bin/cargo-espflash.zip" && \
-    chmod u+x "${HOME}/.cargo/bin/cargo-espflash" && \
-    curl -L "https://github.com/esp-rs/embuild/releases/latest/download/ldproxy-${ARCH}.zip" -o "${HOME}/.cargo/bin/ldproxy.zip" && \
-    unzip "${HOME}/.cargo/bin/ldproxy.zip" -d "${HOME}/.cargo/bin/" && \
-    rm "${HOME}/.cargo/bin/ldproxy.zip" && \
```

Comment:
I see some value on having `espflash` or `cargo-espflash` installed, they offer subcomands like `save-image` that can be really helpful even in a container, and also the user would like to forward the usb-connection and flash from the container as you mentioned.

Reply:
yeah i am on board with it, i just stripped them in here with the mind of a potential container that is often build. If we follow that what i wrote on esp-rs/rust-lang issue i think it should be all included in both the all and the target specific images